jac-uk/admin
# #1347 Enable application process to be configured

## What's included?
Enables an application process to be configured for an exercise so that, for example, minimal information can be requested during initial registration and the remaining information requested when candidates have successfully passed a qualifying test. 
See corresponding functionality in Apply https://github.com/jac-uk/apply/pull/799

To facilitate this functionality a number of fixes and improvements have been made:
- Code is more concise and easier to navigate
- Improved router config
- Improved navigation
  - Less bugs
  - Better UX
  - Applications show in correct status (draft, applied, withdrawn)
- Create exercise 'journey' works as expected
- Exercise overview. Show correct counts of 'draft' and 'applied' applications

## Risk - how likely is this to impact other areas?
🔴 High risk - this includes a lot of changes to shared code

## Who should test?
✅ Developers
✅ Product owner
✅ User testing Group

## How to test?

Below are tests for the expected behavior of Admin.
There are corresponding tests for Apply - see https://github.com/jac-uk/apply/pull/799

### 1. Configure an exercise to have a staged application process
- Find or configure a draft exercise. (if an existing exercise make a change so that this new functionality is activated)
- Navigate to the new 'Application process' page.
- Click to 'Update application process'.
- Drag and drop application 'parts' to the desired stages of the application process.
- Click 'Continue' and notice that the changes have been saved.

**Expected:** the above steps can be carried out successfully

https://user-images.githubusercontent.com/8524401/121855469-4744bc80-cceb-11eb-927f-36b2939dc95e.mov

### 2. Make changes to an exercise and update the application process
- Edit the exercise configuration so that the relevant application 'parts' change. For example make changes to the 'Assessment options'. 
- Navigate to 'Application process' and notice that the new 'parts' have been added to 'Registration'
- Try 'Update application process' again (as above).

**Expected:** the above steps can be carried out successfully

https://user-images.githubusercontent.com/8524401/121855724-968aed00-cceb-11eb-92b3-f13b458acf96.mov


### 3. Progress to the next stage of an application process
- Find an approved exercise which has been configured to have a staged application process
- Navigate to 'Application process' page
- Notice that the current/next stage in the process is indicated
- Click the 'Edit' link to change the current/next stage.
- Fill in the stage, dates and click save
- Notice that the configured stage and dates are updated

**Expected:** the above steps can be carried out successfully

https://user-images.githubusercontent.com/8524401/123976768-53e93600-d9b6-11eb-9150-9248085fe611.mov

---
PREVIEW:DEVELOP
_can be OFF, DEVELOP or STAGING_
